include/WCSimRootEvent.hh

```diff
@@ -87,6 +89,7 @@ public:
   Double_t   GetTime() const { return fTime;}
   Int_t     GetId() const {return fId;}
   Int_t     GetParentId() const {return fParentId;}
+  ProcessType_t GetCreatorProcessName() const {return fCreatorProcess;}
```

Comment:
```suggestion
  ProcessType_t GetCreatorProcess() const {return fCreatorProcess;}
  std::string GetCreatorProcessName() const {return EnumAsString(fCreatorProcess);}
```
This returns the `enum`, not a `string`, so should be named differently. Do also change where you use this (e.g. in the .cc). Note that this is different to `WCSimTrajectory::GetCreatorProcessName()` which returns a `string` and so should not be renamed

Also added a ease-of-use function for getting the string out. I just wrote it in github, so please test it

Reply:
Hi Tom, I don't really understand what I should add this. In my current code, I return every photonCreatorProcess and trackCreatorProcess (or simply creatorProcess) as an enum, and then in my Python pipeline (root->npz->h5) I transform the numbers into strings.

I guess you want me to do this in WCSim directly? This will need changing some files and I found this creatorProcess thing super messy. Please let me know.

Reply:
Adding this function will make your python transform simpler, and more robust against future changes. Doing this, your python analysis script doesn't need to convert the `enum` to a `string` yourself (and is actually an extra layer of complication that can go wrong!). Use `GetProcessName()` on `WCSimRootTrack` & `WCSimRootCherenkovHit` which will do the conversion for you


The point is
- WCSimRootTrack holds an enum
- `GetCreatorProcess()` is a bad name for something that returns an `enum`, so rename that function to be `GetCreatorProcess()`
- Then add a new function `GetCreatorProcessName()` that does actually return a `string`, using the most reliable source of the `enum`->`string` conversion (the WCSim code! Namely `WCSimEnumerations`)

The suggestion I made should work if you click "commit suggestion". However I wrote it on the github website, so haven't tested that it works or compiles - please do so with your analysis pipeline

Reply:
Committing your changes lead to this kind of compilation errors:
```
[  2%] Building CXX object src/CMakeFiles/WCSimRoot.dir/WCSimRootEvent.cc.o
/Users/diiego/Desktop/WCSim/src/WCSimRootEvent.cc:478:34: error: no matching constructor for initialization of 'WCSimRootTrack'
    new(tracks[fNtrack_slots++]) WCSimRootTrack(track->GetIpnu(),
                                 ^              ~~~~~~~~~~~~~~~~~
/Users/diiego/Desktop/WCSim/include/WCSimRootEvent.hh:54:3: note: candidate constructor not viable: no known conversion from 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char>>') to 'ProcessType_t' (aka 'EProcessType') for 13th argument
  WCSimRootTrack(Int_t ipnu,
  ^
/Users/diiego/Desktop/WCSim/include/WCSimRootEvent.hh:26:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 20 were provided
class WCSimRootTrack : public TObject {
      ^
/Users/diiego/Desktop/WCSim/include/WCSimRootEvent.hh:53:3: note: candidate constructor not viable: requires 0 arguments, but 20 were provided
  WCSimRootTrack() {}
```

I guess this is because now in `WCSimRootEvent.cc`, here
```
WCSimRootTrack *WCSimRootTrigger::AddTrack(Int_t ipnu,
					   Int_t flag,
					   Double_t m,
					   Double_t p,
					   Double_t E,
					   Int_t startvol,
					   Int_t stopvol,
					   Double_t dir[3],
					   Double_t pdir[3],
					   Double_t stop[3],
					   Double_t start[3],
					   Int_t parenttype,
                                           ProcessType_t creatorProcess,
					   Double_t time,
					   Int_t id,
					   Int_t idParent,
             std::vector<std::vector<float>> bPs,
             std::vector<float> bKEs,
             std::vector<double> bTimes,
             std::vector<int> bTypes)

```
I should call `ProcessType_t GetCreatorProcess(creatorProcess)` or something like that?

Reply:
Exactly!

Reply:
Sorry @tdealtry but this isn't working 😕 
I changed `ProcessType_t creatorProcess` with `ProcessType_t GetCreatorProcess(creatorProcess)` but compilation says:
```
[  1%] Building CXX object src/CMakeFiles/WCSimRoot.dir/WCSimRootEvent.cc.o
/Users/diiego/Desktop/WCSim/src/WCSimRootEvent.cc:418:46: error: unknown type name 'creatorProcess'
             ProcessType_t GetCreatorProcess(creatorProcess),
                                             ^
/Users/diiego/Desktop/WCSim/src/WCSimRootEvent.cc:406:35: error: out-of-line definition of 'AddTrack' does not match any declaration in 'WCSimRootTrigger'
WCSimRootTrack *WCSimRootTrigger::AddTrack(Int_t ipnu,
                                  ^~~~~~~~
/Users/diiego/Desktop/WCSim/src/WCSimRootEvent.cc:447:13: error: use of undeclared identifier 'creatorProcess'
            creatorProcess,
            ^
/Users/diiego/Desktop/WCSim/src/WCSimRootEvent.cc:478:34: error: no matching constructor for initialization of 'WCSimRootTrack'
    new(tracks[fNtrack_slots++]) WCSimRootTrack(track->GetIpnu(),

```

Reply:
Just pushed a fix (for both these WCSimRootEvent things). It now compiles, please test!